components/expression/expression-renderer.js

```diff
@@ -201,17 +216,24 @@ ExpressionRenderer.prototype.drawMatrixGeneLabels_ = function() {
     return;
   }
   // TODO(liana): Implement the below...
-  var labels = this.svgGeneLabels_.selectAll('text').data([{
-    label: 'a horizontal label'
-  }]);
+  //var labels = this.svgGeneLabels_.selectAll('text').data([{
+  //  label: 'a horizontal label'
+  //}]);
+  var geneLabelsData = this.dataReady_().geneNames;
+  var labels = this.svgGeneLabels_.selectAll('text').data(geneLabelsData);
   labels.enter().append('text')
-    .classed('gene-label', true);
+      .text(function(d) { return d; })
+      .attr('x', 0)
```

Comment:
x, y need to be set in update, instead of enter(). Otherwise this will not handle redraw (e.g. when view is resized).